bin/agent-data-plane/src/cli/run.rs

```diff
@@ -59,24 +59,25 @@ pub async fn run(started: Instant, run_config: RunConfig) -> Result<(), GenericE
                 error!("Failed to create config stream: {}.", e);
                 return Err(e);
             }
-            info!("Waiting for initial configuration from Datadog Agent...");
-
-            // Block until a initial configuration is received.
-            let mut attempts = 0;
-            const CHECK_INTERVAL_MS: u64 = 100;
-
-            while !snapshot_received.load(Ordering::SeqCst) {
-                tokio::time::sleep(Duration::from_millis(CHECK_INTERVAL_MS)).await;
-                attempts += 1;
-
-                if attempts % 100 == 0 {
-                    info!(
-                        "Still waiting for initial configuration... ({}s elapsed)",
-                        attempts / 10
-                    );
+            // If configured, block until a initial configuration is received.
+            if configuration.get_typed_or_default::<bool>("adp.block_until_config_snapshot") {
```

Comment:
I'm realizing now that we should actually include even the config stream creation in this block, and should change the setting to `adp.use_new_config_stream_endpoint`.

Testing locally, with this PR as-is, we still always see at least one error message from the config streamer trying to start the stream but failing to if the Agent isn't new enough to have `StreamConfigEvents`.

I hadn't considered this originally, but really what I was imagining was that we don't use this blocking logic, _or_ the new endpoint, unless explicitly configured to do so.